components/scream/src/mct_coupling/scream_scorpio_interface.F90

```diff
@@ -342,7 +367,11 @@ subroutine register_variable(pio_atm_filename,shortname,longname,numdims,var_dim
     hist_var%compdof(:my_dof_len) = (/ (ii, ii=istart,istop, 1) /)
 
     ! Register Variable with PIO
-    ierr = PIO_def_var(pio_atm_file%pioFileDesc, trim(shortname), hist_var%dtype, hist_var%dimid(:numdims), hist_var%pioVar)
+    ! check to see if variable already is defined with file (for use with input)
+    ierr = PIO_inq_varid(pio_atm_file%pioFileDesc,trim(shortname),hist_var%piovar)
+    
+    ! if ierr is not pio_noerror then the variable needs to be defined
+    if (ierr.ne.pio_noerr) ierr = PIO_def_var(pio_atm_file%pioFileDesc, trim(shortname), hist_var%dtype, hist_var%dimid(:numdims), hist_var%piovar)
     call errorHandle("PIO ERROR: could not define variable "//trim(shortname),ierr)
```

Comment:
Maybe I don't understand the usage of this function correctly, but I think it is being used for both input and output files, which leads to weirdness. I'd think input files should trigger an error if inq_varid isn't defined (which I don't think is done now) and output files should probably trigger an error if the varid is already defined... Maybe register_variable should only be used for outflds and a "get_variable" function should be added to for use by input reads?

Reply:
We could definitely do that to make things clearer. I was hoping to recycle code here since input and output rely on a lot of the same things. But that also means it wouldn’t be hard to create an input specific routine.